0-module-and-module-source.emu

```diff
@@ -0,0 +1,648 @@
+    <emu-clause id="sec-properties-of-the-module-prototype-object">
+        <h1>Properties of the Module Prototype Object</h1>
+        <p>The Module prototype object:</p>
+        <ul>
+            <li>has a [[Prototype]] internal slot whose value is <dfn>%Object.prototype%</dfn>.</li>
+            <li>is %Module.prototype%.</li>
+            <li>is an ordinary object.</li>
+            <li>does not have a [[Module]] or any other of the internal slots that are specific to _Realm_ instance objects.</li>
+        </ul>
+
+        <emu-clause id="sec-module.prototype.constructor">
+            <h1>Module.prototype.constructor</h1>
+            <p>The initial value of `Module.prototype.constructor` is %Module%.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-module.prototype.source">
+            <h1>get Module.prototype.source</h1>
+            <p>`Module.prototype.source` is an accessor property whose set accessor function is *undefined*. Its get accessor function performs the following steps:</p>
+            <emu-alg>
+              1. Let _M_ be the *this* value.
+              1. Perform ? RequireInternalSlot(_M_, [[ModuleSourceInstance]]).
+              1. Return _M_.[[ModuleSourceInstance]].
+            </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-module.prototype-@@tostringtag">
+            <h1>Module.prototype [ @@toStringTag ]</h1>
+            <p>The initial value of the @@toStringTag property is the String value *"Module"*.</p>
+            <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
+        </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-module-instances">
+        <h1>Properties of Module Instances</h1>
+        <p>Module instances are ordinary objects that inherit properties from the Module prototype object (the intrinsic, %Module.prototype%). Module instances are initially created with the internal slots described in <emu-xref href="#table-internal-slots-of-module-instances"></emu-xref>.</p>
+
+        <emu-table id="table-internal-slots-of-module-instances" caption="Internal Slots of Module Instances">
+            <table>
+                <tbody>
+                    <tr>
+                        <th>Internal Slot</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>[[Module]]</td>
+                        <td>Source Text Module Record</td>
+                        <td>The Source Text Module Record that represents this Module Instance.</td>
+                    </tr>
+                    <tr>
+                        <td>[[ModuleSourceInstance]]</td>
+                        <td>ModuleSource Instance or *null*</td>
+                        <td>The Source Text Module Record that represents this Module Instance. *null* if this Module Instance has not a source associated to it.</td>
```

Comment:
What's the purpose of having [[Module]] and [[ModuleSourceInstance]] with the same description but one of them can be null while the other can't? 
This looks like a mistake or a convention I misunderstood.

Reply:
Certainly a mistake in the description. I can fix the description of `[[ModuleSourceInstance]]` to be:

"The ModuleSource Instance associated to Module Instance. *null* if this Module Instance has not a source associated to it."

As for the why this one can be null, there is a note few lines below in the emu-note that describes the scenario where this can be null with an example.